cram_3d_world/cram_bullet_reasoning_belief_state/src/process-modules.lisp

```diff
@@ -0,0 +1,145 @@
+;;; Copyright (c) 2019, Jonas Dech <jdech[at]uni-bremen.de>
+;;; All rights reserved.
+;;; 
+;;; Redistribution and use in source and binary forms, with or without
+;;; modification, are permitted provided that the following conditions are met:
+;;; 
+;;;     * Redistributions of source code must retain the above copyright
+;;;       notice, this list of conditions and the following disclaimer.
+;;;     * Redistributions in binary form must reproduce the above copyright
+;;;       notice, this list of conditions and the following disclaimer in the
+;;;       documentation and/or other materials provided with the distribution.
+;;;     * Neither the name of the Intelligent Autonomous Systems Group/
+;;;       Technische Universitaet Muenchen nor the names of its contributors 
+;;;       may be used to endorse or promote products derived from this software 
+;;;       without specific prior written permission.
+;;; 
+;;; THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+;;; AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+;;; IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+;;; ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+;;; LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+;;; CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+;;; SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+;;; INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+;;; CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+;;; ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+;;; POSSIBILITY OF SUCH DAMAGE.
+
+(in-package :cram-bullet-reasoning-belief-state)
+
+(cpm:def-process-module world-state-detecting (motion-designator)
+  ;;(roslisp:ros-info (world-state-sensing) "Tries to sense the new position of the object")
+  (destructuring-bind (command object) (desig:reference motion-designator)
+    (ecase command
+      (cram-common-designators::sense-pose-object
+       (sense-new-object-pose-from-object object))
+      (cram-common-designators::sense-pose-name
+       (sense-new-object-pose-from-name object)))))
+
+(def-fact-group pr2-matching-pms (cpm:matching-process-module)
+  (<- (cpm:matching-process-module ?motion-designator world-state-detecting)
+    (desig:desig-prop ?motion-designator (:type :world-state-detecting))))
+
+(defun sense-new-object-pose-from-name (?old-object)
+  (let (map->base
+        map->obj
+        trans
+        ?new-pose
+        ?new-transform
+        ?new-pose-in-map
+        ?new-transform-in-map
+        new-object
+        ?type)
+   
+    (setf map->base (cl-tf:pose->transform (cram-tf:robot-current-pose)))
```

Comment:
Bitte benutze nicht ``cl-tf``, wenn es nicht noetig ist. ``cl-tf`` ist ein Package, dass ein TF Listener erstellt. In diesem Fall brauchst du kein TF listener, also reicht ``cl-transforms`` aus.